rules/DowngradePhp71/Rector/StaticCall/DowngradeClosureFromCallableRector.php

```diff
@@ -0,0 +1,116 @@
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Expr\ClosureUse;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\Node\Stmt\Return_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Naming\Naming\VariableNaming;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\PostRector\Collector\NodesToAddCollector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://wiki.php.net/rfc/closurefromcallable
+ *
+ * @see \Rector\Tests\DowngradePhp71\Rector\StaticCall\DowngradeClosureFromCallableRector\DowngradeClosureFromCallableRectorTest
+ */
+final class DowngradeClosureFromCallableRector extends AbstractRector
+{
+    public function __construct(
+        private NodesToAddCollector $nodesToAddCollector,
+        private VariableNaming $variableNaming
+    ) {
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [StaticCall::class];
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Converts Closure::fromCallable() to compatible alternative.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+\Closure::fromCallable(callable);
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$callable = callable;
+function () use ($callable) {
+    return $callable(...func_get_args());
+};
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @param StaticCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (!$this->isClosureFromCallable($node)) {
+            return null;
+        }
+
+        $tempVariableName = $this->variableNaming->createCountedValueName(
+            'callable',
+            $node->getAttribute(AttributeKey::SCOPE)
+        );
+
+        $tempVariable = new Variable($tempVariableName);
+
+        $assignment = new Expression(
+            new Assign(
+                $tempVariable,
+                $node->args[0]->value
+            )
+        );
+
+        $currentStatement = $node->getAttribute(AttributeKey::CURRENT_STATEMENT);
+        $this->nodesToAddCollector->addNodeBeforeNode($assignment, $currentStatement);
+
+        $closure = new Closure;
+        $closure->uses[] = new ClosureUse($tempVariable);
+
+        $innerFuncCall = new FuncCall(
+            $tempVariable,
+            [
+                new Arg(
+                    $this->nodeFactory->createFuncCall('func_get_args'),
+                    false,
+                    true
+                )
+            ]
+        );
+
+        $closure->stmts[] = new Return_($innerFuncCall);
+
+        return $closure;
+    }
+
+    private function isClosureFromCallable(StaticCall $node): bool
```

Comment:
I think it can be renamed to `shouldSkip(StaticCall $staticCall): bool`, so can be used with `if ($this->shouldSkip($node))`:

```php
private function shouldSkip(StaticCall $node): bool
{
        if (! $this->nodeNameResolver->isName($node->class, 'Closure')) {
                return true;
        }

        return ! $this->nodeNameResolver->isName($node->name, 'fromCallable');
}
```